### PR TITLE
feat: Submit Coin98 Wallet

### PR DESCRIPTION
**Pull request by khiemsoft**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

# Type of change.

- [ ] FIX - a PR of this type patches a bug.
- [x] FEATURE - a PR of this type introduces a new feature.
- [ ] BUILD - a PR of this type introduces build changes.
- [ ] CI - a PR of this type introduces CI changes.
- [ ] DOCS - a PR of this type introduces DOCS improvement.
- [ ] STYLE - a PR of this type introduces style changes.
- [ ] REFACTOR - a PR of this type introduces refactoring.
- [ ] PERFORMANCE - a PR of this type introduces performance changes.
- [ ] TEST - a PR of this type adds more tests.
- [ ] CHORE - a PR introduces other changes than the specified above.

### CODE REVIEW

File: packages/coin98-wallet/src/lib/coin98-wallet.ts [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+
+    async signOut() {
+      // Ignore if unsuccessful (returns false).
+      await _state.wallet.near.disconnect();
+    },
+
+    async getAccounts() {
+      return getAccounts();
+    },
+
```
Comment (amirsaran3) [inline packages/coin98-wallet/src/lib/coin98-wallet.ts:104]: The `BaseWalletBehaviour` has a new `verifyOwner(params: VerifyOwnerParams): Promise<VerifiedOwner | void>;` function so you have to add it in this return object. You can check out for example the `math-wallet` package and copy how they did it.

File: packages/coin98-wallet/src/lib/coin98-wallet.ts [hunk 2]
```diff
@@ -0,0 +1,167 @@
...
+    if (mobile) {
+      return null;
+    }
+
+    return {
+      id: "coin98-wallet",
+      type: "injected",
+      metadata: {
+        name: "Coin98 Wallet",
+        description: "Crypto Wallet & Defi Gateway",
```
Comment (amirsaran3) [inline packages/coin98-wallet/src/lib/coin98-wallet.ts:157]: We are taking wallet descriptions from [awesomenear](https://awesomenear.com/). Can you change this to the one that is on there.

File: packages/coin98-wallet/project.json [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+      "executor": "@nrwl/web:rollup",
+      "outputs": ["{options.outputPath}"],
+      "options": {
+        "outputPath": "dist/packages/coin98-wallet",
+        "tsConfig": "packages/coin98-wallet/tsconfig.lib.json",
+        "project": "packages/coin98-wallet/package.json",
+        "entryFile": "packages/coin98-wallet/src/index.ts",
+        "buildableProjectDepsInPackageJsonType": "dependencies",
+        "compiler": "babel",
+        "format": ["esm", "umd", "cjs"],
```
Comment (amirsaran3) [inline packages/coin98-wallet/project.json:16]: You will need to remove "umd" from this list for build to work.

File: packages/coin98-wallet/src/lib/coin98-wallet.ts (hunk 1 above)
Comment (khiemsoft) [reply packages/coin98-wallet/src/lib/coin98-wallet.ts:108]: commented fixed

File: packages/coin98-wallet/src/lib/coin98-wallet.ts [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
...cxNyAxMS43MTcgMCAwIDAgMy40NDUtOC4yODQgMTEuNzAzIDExLjcwMyAwIDAgMC0xLjk3NC02LjUxMiAxMS43MzYgMTEuNzM2IDAgMCAwLTUuMjY2LTQuMzJabS00LjUxIDMuOTE3YTYuOTQ1IDYuOTQ1IDAgMCAxIDQuODk3IDIuMDI5IDYuOTE4IDYuOTE4IDAgMCAxIDIuMDMyIDQuODg2IDYuOTA2IDYuOTA2IDAgMCAxLTEuMTY4IDMuODQyIDYuOTQgNi45NCAwIDAgMS0xMC42NiAxLjA0OCA2LjkxMSA2LjkxMSAwIDAgMS0xLjUtNy41MzYgNi45MTggNi45MTggMCAwIDEgMi41NS0zLjEwMyA2LjkzNyA2LjkzNyAwIDAgMSAzLjg1LTEuMTY2Wm0yNC41Ni00LjgxYTEwLjA1OSAxMC4wNTkgMCAwIDAtNy4xMDMgMi45NCAxMC4wMiAxMC4wMiAwIDAgMC0yLjk0IDcuMDkgOS45IDkuOSAwIDAgMCAxLjIzIDQuNzk1IDEzLjU3NSAxMy41NzUgMCAwIDEgNC4yMTQtMi4zMjIgNS4wODIgNS4wODIgMCAwIDEtLjYyNS0yLjQ3NyA1LjIwNiA1LjIwNiAwIDAgMSAxLjUwMy0zLjczNiA1LjIyMyA1LjIyMyAwIDAgMSAzLjcyMi0xLjU1NCA1LjIzNCA1LjIzNCAwIDAgMSAzLjcyIDEuNTU0IDUuMjEzIDUuMjEzIDAgMCAxIDEuNTA1IDMuNzM2IDUuMjc5IDUuMjc5IDAgMCAxLS42MjMgMi40NzMgMTMuNTc0IDEzLjU3NCAwIDAgMSA0LjIxMyAyLjMyMiA5LjkwMyA5LjkwMyAwIDAgMCAxLjIzLTQuNzk1IDEwLjAzMiAxMC4wMzIgMCAwIDAtMi45NDYtNy4wODYgMTAuMDcgMTAuMDcgMCAwIDAtNy4xLTIuOTRabS0yMy43NSA3Ljk5aC0xLjYxN3YxLjc1YTIuNDE5IDIuNDE5IDAgMCAwLTEuNTgyIDIuNjg3IDIuNDE0IDIuNDE0IDAgMCAwIDIuMzkgMi4wMDYgMi40NSAyLjQ1IDAgMCAwIDEuNTU1LS41NzQgMi40MTQgMi40MTQgMCAwIDAtLjc0Ni00LjExOXYtMS43NVoiLz4KICA8L2c+Cjwvc3ZnPgo=",
```
[the remark this answers is not in the document]
Comment (khiemsoft) [reply packages/coin98-wallet/src/lib/coin98-wallet.ts]: commented fixed

File: packages/coin98-wallet/src/lib/coin98-wallet.ts (hunk 2 above)
Comment (khiemsoft) [reply packages/coin98-wallet/src/lib/coin98-wallet.ts]: commented fixed

File: packages/coin98-wallet/project.json (hunk 3 above)
Comment (khiemsoft) [reply packages/coin98-wallet/project.json]: commented fixed

File: packages/coin98-wallet/src/lib/coin98-wallet.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (amirsaran3) [reply packages/coin98-wallet/src/lib/coin98-wallet.ts]: Can you look at the other packages and how we import icons and change accordingly.

File: examples/react/contexts/WalletSelectorContext.tsx [hunk 5]
```diff
@@ -43,6 +44,7 @@ export const WalletSelectorContextProvider: React.FC = ({ children }) => {
       modules: [
         ...(await setupDefaultWallets()),
         setupNearWallet(),
+        setupCoin98Wallet(),
```
Comment (amirsaran3) [inline examples/react/contexts/WalletSelectorContext.tsx:47]: Can you also add Coin98 to the angular example as well.

Comment (khiemsoft) [reply examples/react/contexts/WalletSelectorContext.tsx:47]: commented fixed

File: examples/angular/src/app/app.component.ts [hunk 6]
```diff
@@ -59,7 +63,7 @@ export class AppComponent implements OnInit {
           },
         }),
         setupNightlyConnect({
-          url: "wss://ncproxy.nightly.app/app",
```
Comment (amirsaran3) [inline examples/angular/src/app/app.component.ts:62]: Why have you changed this?

Comment (khiemsoft) [reply examples/angular/src/app/app.component.ts]: Sorry my mistake not synching to latest dev. Just fixed it

File: examples/angular/src/app/app.component.ts [hunk 7]
```diff
@@ -46,12 +45,12 @@ export class AppComponent implements OnInit {
       network: "testnet",
       debug: true,
       modules: [
-        ...(await setupDefaultWallets()),
         setupNearWallet(),
+        setupMyNearWallet(),
         setupSender(),
         setupMathWallet(),
         setupNightly(),
-        setupMeteorWallet(),
```
Comment (amirsaran3) [inline examples/angular/src/app/app.component.ts:51]: The other wallets need to be in here.

File: packages/coin98-wallet/src/lib/coin98-wallet.ts [hunk 8]
```diff
@@ -0,0 +1,206 @@
...
+        results.push(await provider.sendTransaction(signedTransactions[i]));
+      }
+
+      return results;
+    },
+  };
+};
+
+export const setupCoin98Wallet = ({
+  iconUrl = defaultCoin98Logo
```
Comment (amirsaran3) [inline packages/coin98-wallet/src/lib/coin98-wallet.ts:181]: Can you change this so it is like the other packages. Using a string path and assets folder.

File: packages/coin98-wallet/src/lib/icons.ts [hunk 9]
```diff
@@ -0,0 +1 @@
...jcxNyAxMS43MTcgMCAwIDAgMy40NDUtOC4yODQgMTEuNzAzIDExLjcwMyAwIDAgMC0xLjk3NC02LjUxMiAxMS43MzYgMTEuNzM2IDAgMCAwLTUuMjY2LTQuMzJabS00LjUxIDMuOTE3YTYuOTQ1IDYuOTQ1IDAgMCAxIDQuODk3IDIuMDI5IDYuOTE4IDYuOTE4IDAgMCAxIDIuMDMyIDQuODg2IDYuOTA2IDYuOTA2IDAgMCAxLTEuMTY4IDMuODQyIDYuOTQgNi45NCAwIDAgMS0xMC42NiAxLjA0OCA2LjkxMSA2LjkxMSAwIDAgMS0xLjUtNy41MzYgNi45MTggNi45MTggMCAwIDEgMi41NS0zLjEwMyA2LjkzNyA2LjkzNyAwIDAgMSAzLjg1LTEuMTY2Wm0yNC41Ni00LjgxYTEwLjA1OSAxMC4wNTkgMCAwIDAtNy4xMDMgMi45NCAxMC4wMiAxMC4wMiAwIDAgMC0yLjk0IDcuMDkgOS45IDkuOSAwIDAgMCAxLjIzIDQuNzk1IDEzLjU3NSAxMy41NzUgMCAwIDEgNC4yMTQtMi4zMjIgNS4wODIgNS4wODIgMCAwIDEtLjYyNS0yLjQ3NyA1LjIwNiA1LjIwNiAwIDAgMSAxLjUwMy0zLjczNiA1LjIyMyA1LjIyMyAwIDAgMSAzLjcyMi0xLjU1NCA1LjIzNCA1LjIzNCAwIDAgMSAzLjcyIDEuNTU0IDUuMjEzIDUuMjEzIDAgMCAxIDEuNTA1IDMuNzM2IDUuMjc5IDUuMjc5IDAgMCAxLS42MjMgMi40NzMgMTMuNTc0IDEzLjU3NCAwIDAgMSA0LjIxMyAyLjMyMiA5LjkwMyA5LjkwMyAwIDAgMCAxLjIzLTQuNzk1IDEwLjAzMiAxMC4wMzIgMCAwIDAtMi45NDYtNy4wODYgMTAuMDcgMTAuMDcgMCAwIDAtNy4xLTIuOTRabS0yMy43NSA3Ljk5aC0xLjYxN3YxLjc1YTIuNDE5IDIuNDE5IDAgMCAwLTEuNTgyIDIuNjg3IDIuNDE0IDIuNDE0IDAgMCAwIDIuMzkgMi4wMDYgMi40NSAyLjQ1IDAgMCAwIDEuNTU1LS41NzQgMi40MTQgMi40MTQgMCAwIDAtLjc0Ni00LjExOXYtMS43NVoiLz4KICA8L2c+Cjwvc3ZnPgo="
```
Comment (amirsaran3) [inline packages/coin98-wallet/src/lib/icons.ts:1]: Remove this

Comment (khiemsoft) [reply packages/coin98-wallet/src/lib/icons.ts]: commented fixed

Comment (amirsaran3) [reply packages/coin98-wallet/src/lib/icons.ts]: > commented fixed

You need to create an `assets` folder and place your logo image inside. Then reference the image with a path. Check out other packages and see how it is done.

Comment (khiemsoft) [reply packages/coin98-wallet/src/lib/icons.ts]: Commented Fixed

**Comment by AmmarHumackicSQA** [comment]
Hey @khiemsoft, as a reminder, the security statement needs to be shown either on your GitHub repo or the website. You can check it [here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md#:~:text=GitHub%20account%20or%20Website) again. Thanks!

**Comment by AmmarHumackicSQA** [comment]
> Hey @khiemsoft, as a reminder, the security statement needs to be shown either on your GitHub repo or the website. You can check it [here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md#:~:text=GitHub%20account%20or%20Website) again. Thanks!

Hi @khiemsoft can we get a response on this so the PR can be merged, as we think it's ready? Thanks.

**Comment by khiemsoft** [comment]
> 

@AmmarHumackicSQA Thanks for your review. We will add a security statement on our Github repo tomorrow.

**Comment by khiemsoft** [comment]
> Hey @khiemsoft, as a reminder, the security statement needs to be shown either on your GitHub repo or the website. You can check it [here](https://github.com/near/wallet-selector/blob/main/CONTRIBUTING.md#:~:text=GitHub%20account%20or%20Website) again. Thanks!

https://github.com/coin98/near-wallet-security-criteria/tree/main
Here is our near wallet security criteria

**Comment by AmmarHumackicSQA** [comment]
Thank you @khiemsoft !

**Comment by amirsaran3** [comment]
Hey @khiemsoft. There were some changes in the meantime so some conflicts have appeared in this PR. You will have to resolve them before we can merge into dev.

**Comment by khiemsoft** [comment]
> Hey @khiemsoft. There were some changes in the meantime so some conflicts have appeared in this PR. You will have to resolve them before we can merge into dev.

Hi I just sync to latest dev branch. Thanks for your review

**Comment by amirsaran3** [comment]
Hey @khiemsoft. 

There were some changes to the iconUrl code. When you run the development server you will see that there is no icon showing. Can you fix that by looking at other wallet packages and how they did it.

Currently when I import my testnet account and try to log in it does not work. I select the `NEAR NEP141` chain, import 12 word phrase (also tried with private key) and it is successful. The account id of my account is `amirsaran.testnet`, but it generates a completely new account id (this one `9e89ee7c80656a77f59373c61cdc3752b85f74025449e478410a9c633f258f54`) and uses that when logging in.

I am getting this error when trying to log in:
```
Error: [-32000] Server error: account 9e89ee7c80656a77f59373c61cdc3752b85f74025449e478410a9c633f258f54 does not exist while viewing
```

Can you explain or show a quick video of how you are correctly using the Coin98 wallet?

**Comment by khiemsoft** [comment]
> Hey @khiemsoft.
> 
> There were some changes to the iconUrl code. When you run the development server you will see that there is no icon showing. Can you fix that by looking at other wallet packages and how they did it.
> 
> Currently when I import my testnet account and try to log in it does not work. I select the `NEAR NEP141` chain, import 12 word phrase (also tried with private key) and it is successful. The account id of my account is `amirsaran.testnet`, but it generates a completely new account id (this one `9e89ee7c80656a77f59373c61cdc3752b85f74025449e478410a9c633f258f54`) and uses that when logging in.
> 
> I am getting this error when trying to log in:
> 
> ```
> Error: [-32000] Server error: account 9e89ee7c80656a77f59373c61cdc3752b85f74025449e478410a9c633f258f54 does not exist while viewing
> ```
> 
> Can you explain or show a quick video of how you are correctly using the Coin98 wallet?

For the iconUrl just fixed at the latest commit.
Currently Coin98 Wallet not supported for account ID.

**Comment by amirsaran3** [comment]
@khiemsoft Even when I create a new account trough Coin98 it does not work. Can you explain how you used the Coin98 wallet to test on our example dApp.

**Comment by khiemsoft** [comment]
> @khiemsoft Even when I create a new account trough Coin98 it does not work. Can you explain how you used the Coin98 wallet to test on our example dApp.

You need to deposit minimum Near require to create account on Near chain. Then you can interact with Dapps Near via Coin98 Extension

**Review by amirsaran3 — APPROVED** [review]

**Comment by khiemsoft** [comment]
> @khiemsoft There are still conflicts, can you resolve those to be able to merge.

Commented fixed. Thanks for your review

**Comment by amirsaran3** [comment]
Hey @khiemsoft, as you can see at the bottom of this page it still says: "This branch has conflicts that must be resolved". Can you look again and resolve the conflicts. Your branch is 230 commits behind of `near:dev`.

**Comment by khiemsoft** [comment]
> Hey @khiemsoft, as you can see at the bottom of this page it still says: "This branch has conflicts that must be resolved". Can you look again and resolve the conflicts. Your branch is 230 commits behind of `near:dev`.

conflict resolved done. Thanks for your review

**Review by amirsaran3 — APPROVED** [review]

**Pull request merged**